### PR TITLE
Feature post dashboard items in 2.29

### LINKED ISSUE #119: Fix dashboard push (update dashboard name, add/remove dashboardItems) on api 2.29

### LINKED ISSUE #123: fix dashboard push (remove dashboard and create new dashboards in api 2.29)

### PR DESCRIPTION
**Pull request by idelcano**
Closes #119
The push of dashboard and dashboarditem, content, interpretations, comments in 2.29=> is different than 2.28<=. 
The objects have new fields or modified fields like access.

To review-> Push and deletion of dashboard, dashboarditems, interpretations and coments.
The user can't update dashboards (editing name) in 2.29=>.

Pull request final features:

Create a dashboard and push in 2.29=>
Delete a dashboard in 2.29=>.
Create and pull shared interpretations and comments in 2.29=>.
Allow add new dashboarditems into dashboard, and post using api/26/ call in all versions.
Disable dashboard update(disabling the edit name edittext)), to prevent override all the dashboarditems unknown fields in 2.29=>.

This branch starts from https://github.com/EyeSeeTea/dhis2-android-dashboard/pull/125

This branch was merged:
#122
#121

### CODE REVIEW

File: api/src/main/java/org/hisp/dhis/android/dashboard/api/controllers/DashboardController.java [hunk 1]
```diff
@@ -368,7 +367,12 @@ private void sendDashboardChanges() throws APIException {
 
     private void postDashboard(Dashboard dashboard) throws APIException {
         try {
-            Response response = mDhisApi.postDashboard(dashboard);
+            Response response;
+            if (SystemInfo.isLoggedInServerWithLatestApiVersion()) {
+                response = mDhisApi.postDashboard(Dashboard.createNewApiDashboard(dashboard));
```
Comment (xurxodev) [inline api/src/main/java/org/hisp/dhis/android/dashboard/api/controllers/DashboardController.java]: Dashboard.createNewApiDashboard create a new dashboard and set all fields from old dashboard, is it necessary this?

File: api/src/main/java/org/hisp/dhis/android/dashboard/api/models/Access.java [hunk 2]
```diff
@@ -140,6 +153,7 @@ public String toString() {
                 .append(", read=").append(read)
                 .append(", update=").append(update)
                 .append(", delete=").append(delete)
+                .append(", data=").append(data.toString())
```
Comment (xurxodev) [inline api/src/main/java/org/hisp/dhis/android/dashboard/api/models/Access.java]: if data is null, this throw NullPointException

File: api/src/main/java/org/hisp/dhis/android/dashboard/api/models/Dashboard.java [hunk 3]
```diff
@@ -285,4 +287,16 @@ public void setCreated(String created) {
     public void setLastUpdated(String lastUpdated) {
         this.lastUpdated = lastUpdated;
     }
+    //creates a new dashboard with the new api required fields
+    public static Dashboard createNewApiDashboard(Dashboard oldDashboard){
```
Comment (xurxodev) [inline api/src/main/java/org/hisp/dhis/android/dashboard/api/models/Dashboard.java]: createNewApiDashboard create a new dashboard and set all fields from old dashboard, is it necessary this?

**Review by xurxodev — CHANGES_REQUESTED** [review]
@idelcano I have found something to review, can you 
 take a look?

**Comment by idelcano** [comment]
@xurxodev  done, i tested without that code and works.

**Review by xurxodev — APPROVED** [review]
thanks @idelcano

**Pull request merged**